### PR TITLE
fix: enable optimize upgrade process as initContainer

### PR DESCRIPTION
**Pull request by jessesimpson36**
### Which problem does the PR fix?

This PR will run the optimize upgrade process as an initContainer.  This was a recommended fix inside our documentation  for a very long time, but I'm surprised that it hasn't been brought up to our team.
https://docs.camunda.org/optimize/3.6/technical-guide/update/#3-2-automatic-update-execution-optimize-3-2-0

https://docs.camunda.io/optimize/self-managed/optimize-deployment/migration-update/instructions/

#838 

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/CONTRIBUTING.md).

**Before opening the PR:**

- [x] There is no other open [pull request](../pulls) for the same update/change.
- [x] The commits follow our [Commit Guidelines](../blob/main/CONTRIBUTING.md#commit-guidelines).
- [x] Tests for charts are added (if needed).
- [x] The main Helm chart and sub-chart are updated (if needed).
- [x] In-repo [documentation](../blob/main/CONTRIBUTING.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [x] Did all checks/tests pass in the PR?

### CODE REVIEW

File: charts/camunda-platform/charts/optimize/templates/deployment.yaml [hunk 1]
```diff
@@ -160,3 +160,16 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
 {{- end }}
+      initContainers:
```
Comment (aabouzaid) [inline charts/camunda-platform/charts/optimize/templates/deployment.yaml:163]: Please add a flag to enable/disable it.

**Comment by aabouzaid** [comment]
@jessesimpson36 That's a good addition, thanks Jesse :ok_hand: 
I've left a minor comment.

File: charts/camunda-platform/charts/optimize/templates/deployment.yaml (hunk 1 above)
Comment (aabouzaid) [reply charts/camunda-platform/charts/optimize/templates/deployment.yaml:163]: And let's have it disabled by default and enable it with 8.3.

Comment (jessesimpson36) [reply charts/camunda-platform/charts/optimize/templates/deployment.yaml:164]: done

**Comment by jessesimpson36** [comment]
I've added this to the 8.3 checklist as a "Nice to have" and put a note to enable it by default for 8.3

**Comment by aabouzaid** [comment]
The unit test fails because it checks out the main branch instead of the PR branch (I will check the issue later).
So this PR is ready to be merged.

File: charts/camunda-platform/test/unit/optimize/golden/deployment.golden.yaml [hunk 2]
```diff
@@ -98,4 +98,17 @@ spec:
...
+      - name: migration
+        image: "camunda/optimize:3.10.5"
+        command: ['./upgrade/upgrade.sh', '--skip-warning']
+        env:
+        - name: CAMUNDA_OPTIMIZE_ZEEBE_ENABLED
+          value: "true"
+        - name: CAMUNDA_OPTIMIZE_ZEEBE_PARTITION_COUNT
+          value: "3"
+        - name: OPTIMIZE_ELASTICSEARCH_HOST
+          value: "elasticsearch-master"
```
Comment (lukas-beumer) [inline charts/camunda-platform/test/unit/optimize/golden/deployment.golden.yaml:112]: This is not working with external Elasticsearch clusters anymore. If it's not hosted in the Kubernetes cluster directly

Comment (aabouzaid) [reply charts/camunda-platform/test/unit/optimize/golden/deployment.golden.yaml:112]: Just for records, follow-up issue: https://github.com/camunda/camunda-platform-helm/issues/953

**Pull request merged**